### PR TITLE
Supplier can retire a listing 26

### PR DESCRIPTION
**Pull request by scottalexandra**
Added tests to make sure a supplier can retire a listing and that listing does not appear on items index page if it is retired. Also fixed a bug I found on items controller. When no category was selected, app was breaking instead of rendering flash message and redirecting back... switch up the order the methods were called to make sure app does not break....

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -27,4 +27,10 @@ def item_supplier
     supplier_id = self.supplier_id
     Supplier.find(supplier_id).slug
   end
+
+  def add_categories_to_item(categories)
+    categories.each do |cat_name|
+      self.categories << Category.find_or_create_by(name: cat_name)
```
Comment (bmrsny) [inline app/models/item.rb]: Should this just be find category since it is a check box?

Comment (scottalexandra) [inline app/models/item.rb]: I just changed that. I agree, I think find_by is all that necessary..

**Comment by scottalexandra** [comment]
Added category name uniquess validation.... Sadly this pull request doesn't show the tests because there is too much seed data...

**Comment by bmrsny** [comment]
This looks good.

**Pull request merged**